components/encryption/src/config.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Default, Configuration)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    #[config(skip)]
+    pub encryption_dir: String,
+
+    #[config(skip)]
+    pub kms: KmsConfig,
+    // TODO add file backend config
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Default, Configuration)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+pub struct KmsConfig {
+    #[config(skip)]
+    pub key_id: String,
+
+    #[config(skip)]
+    pub access_key: String,
+    #[config(skip)]
+    pub secret_access_key: String,
+
+    #[config(skip)]
+    pub region: String,
+    #[config(skip)]
+    pub endpoint: String,
```

Comment:
remove `access_key` and `secret_access_key` (we should be getting credentials from env vars, aws config, or IAM), and make `region` and `endpoint` optional?